Import content drive and pandas for notebook access.

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import pandas as pd

(2) Use pandas to read the olympic_medals.csv from the DATA folder on Google Drive. Then perform the followings by
using pandas and/or Python built-in functions.

2-a) Find the size of the data frame. How many rows, how many columns?


In [ ]:
df = pd.read_csv('/content/drive/MyDrive/DATA/olympic_medals.csv')

display(df)
df.shape

Gender  \
0       M     
1       M     
2       M     
3       M     
4       M     
...     ...   
2389    W     
2390    W     
2391    W     
2392    W     
2393    W     

           Event         \
0            10000M Men   
1            10000M Men   
2            10000M Men   
3            10000M Men   
4            10000M Men   
...                 ...   
2389  Triple Jump Women   
2390  Triple Jump Women   
2391  Triple Jump Women   
2392  Triple Jump Women   
2393  Triple Jump Women   

     Location  \
0         Rio   
1         Rio   
2         Rio   
3     Beijing   
4     Beijing   
...       ...   
2389   Athens   
2390   Athens   
2391  Atlanta   
2392  Atlanta   
2393  Atlanta   

      Year  \
0     2016   
1     2016   
2     2016   
3     2008   
4     2008   
...    ...   
2389  2004   
2390  2004   
2391  1996   
2392  1996   
2393  1996   

     Medal  \
0       G    
1       S    
2       B    
3       G    
4       S    
...    ...   
2389    S    
2390    B    
2391    G    
2392    S    
2393    B    

              Name           \
0             Mohamed FARAH   
1     Paul Kipngetich TANUI   
2              Tamirat TOLA   
3           Kenenisa BEKELE   
4            Sileshi SIHINE   
...                     ...   
2389      Hrysopiyi DEVETZI   
2390       Tatyana LEBEDEVA   
2391         Inessa KRAVETS   
2392        Inna LASOVSKAYA   
2393       Sarka KASPARKOVA   

     Nationality  \
0         GBR      
1         KEN      
2         ETH      
3         ETH      
4         ETH      
...          ...   
2389      GRE      
2390      RUS      
2391      UKR      
2392      RUS      
2393      CZE      

       Result   
0     25:05.17  
1     27:05.64  
2     27:06.26  
3     27:01.17  
4     27:02.77  
...        ...  
2389     15.25  
2390     15.14  
2391     15.33  
2392     14.98  
2393     14.98  

[2394 rows x 8 columns]

(2394, 8)


    There are 2,394 rows and 8 columns in the dataframe containing the olympic medal csv.

2-b) Use the appropriate function to report the data types for all columns.

In [ ]:
df.dtypes

Gender         object
Event          object
Location       object
Year            int64
Medal          object
Name           object
Nationality    object
Result         object
dtype: object

    The data types for all the columns are **object**, except for the *year*, which is a **64-bit integer**.

2-c) Find how many unique locations olympics were held at.


In [ ]:
df['Location'].unique() #this only prints array of unique locations

len(df['Location'].unique()) #this prints length of the above array


23

    There are 23 unique locations for the Olympics.

2-d) Find how many medals the USA team won in total.



In [ ]:
len(df[df['Nationality']=='USA'])

638

    There have been 638 medals won by the USA team in total.

2-e) Find the total number of medals for each Nationality. (Hint: please search for groupby() function with count)

    **This block below sets pandas display() option to show entire dataframe.

In [ ]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('display.width', 1000)
pd.set_option('display.colheader_justify', 'center')
pd.set_option('display.precision', 3)

In [ ]:
display(df.groupby('Nationality').count().sort_values('Medal', ascending = False).Medal)

(3) Use pandas to read the *causes_of_death.csv* from the DATA folder on Google Drive. Then perform the followings
by using pandas and/or Python built-in functions.

3-a) Find the size of the data frame. How many rows, how many columns?

In [ ]:
pd.reset_option('display') #limits pandas display to show 5 at head and tail of data frame, instead of all elements like previously

In [ ]:
cd = pd.read_csv('/content/drive/MyDrive/DATA/causes_of_death.csv')

display(cd)
cd.shape

State  State Code Ten-Year Age Groups  ... Deaths Population Crude Rate
0     Alabama           1            < 1 year  ...     14     3579.0        NaN
1     Alabama           1            < 1 year  ...     24     7443.0      322.5
2     Alabama           1            < 1 year  ...   2093   169339.0     1236.0
3     Alabama           1            < 1 year  ...   2144   347921.0      616.2
4     Alabama           1            < 1 year  ...     33     7366.0      448.0
...       ...         ...                 ...  ...    ...        ...        ...
4110  Wyoming          56           85+ years  ...  13140    95575.0    13748.4
4111  Wyoming          56           85+ years  ...     41      356.0    11516.9
4112  Wyoming          56           85+ years  ...     12      244.0        NaN
4113  Wyoming          56           85+ years  ...     32      454.0     7048.5
4114  Wyoming          56           85+ years  ...   7982    49954.0    15978.7

[4115 rows x 11 columns]

(4115, 11)

    There are 4,115 rows and 11 columns in the dataframe containing the cause of death csv.

3-b) What is the average death count across the country?

In [ ]:
from statistics import mean
mean(cd.Deaths)

10914.44957472661

3-c) What is the total death count for each state. (Hint: please search for groupby() function with sum)

In [ ]:
cd2 = cd[['State', 'Deaths']]
display(cd2.groupby('State').sum('Death'))

#cd2 = cd.loc[:,['State', 'Deaths']]    this works as well inst. of 1st

3-d) What is the average death count for each state. (Hint: please search for groupby() function with mean)

In [ ]:
display(cd2.groupby('State').mean('Death'))

Deaths
State                             
Alabama               10626.913580
Alaska                  772.365854
Arizona                9109.717391
Arkansas               6455.728395
California            45819.797872
Colorado               6378.215909
Connecticut            6559.135802
Delaware               1970.043478
District of Columbia   1447.276923
Florida               34790.122222
Georgia               14639.732558
Hawaii                 2266.368421
Idaho                  2778.847222
Illinois              21519.206897
Indiana               12895.425000
Iowa                   6185.939024
Kansas                 5407.566265
Kentucky               9902.434211
Louisiana              9107.397590
Maine                  3310.800000
Maryland               9201.264368
Massachusetts         12185.567901
Michigan              17976.876404
Minnesota              7980.500000
Mississippi            6117.755814
Missouri              11862.211765
Montana                2200.402778
Nebraska               3474.525000
Nevada                 4014.476744
New Hampshire          3030.919355
New Jersey            15375.642857
New Mexico             3236.758621
New York              29799.728261
North Carolina        15373.725275
North Dakota           1650.476923
Ohio                  24534.271605
Oklahoma               7376.797753
Oregon                 6544.909091
Pennsylvania          27221.823529
Rhode Island           2448.138889
South Carolina         9044.086420
South Dakota           1863.753623
Tennessee             13169.469136
Texas                 32991.266667
Utah                   3028.825581
Vermont                1944.897959
Virginia              12601.917647
Washington             9726.808989
West Virginia          6229.419355
Wisconsin              9745.784091
Wyoming                1155.014925

3-e) What is the total death count for males and females in each state. (Hint: groupby() function with sum, but you
will have two attributes/columns for the groupby() )

In [ ]:
display(cd.groupby(['State','Gender']).sum('Death'))


State Code  Deaths  Population  Crude Rate
State         Gender                                            
Alabama       Female          40  430133  43148507.0     69294.4
              Male            41  430647  40597842.0     86614.7
Alaska        Female          80   27199   5761611.0     77863.9
              Male            84   36135   6379314.0    100396.4
Arizona       Female         180  396028  54846072.0     73199.1
...                          ...     ...         ...         ...
West Virginia Male          1674  192560  16087802.0     76930.9
Wisconsin     Female        2420  436397  50807788.0     83793.6
              Male          2420  421232  49954108.0    107111.4
Wyoming       Female        1792   37153   4696239.0     66382.2
              Male          1960   40233   4872826.0     68545.2

[102 rows x 4 columns]

3-f) Drop your rows which have NA in it. Then verify it by using head() to show they are removed. (Hint: dropna())




In [ ]:
cd.dropna()
cd.head()

State  State Code Ten-Year Age Groups  ... Deaths Population Crude Rate
0  Alabama           1            < 1 year  ...     14     3579.0        NaN
1  Alabama           1            < 1 year  ...     24     7443.0      322.5
2  Alabama           1            < 1 year  ...   2093   169339.0     1236.0
3  Alabama           1            < 1 year  ...   2144   347921.0      616.2
4  Alabama           1            < 1 year  ...     33     7366.0      448.0

[5 rows x 11 columns]

3-g) Create a column called ratio and assign the ratio of death counts to population to fill its values. Then verify it
by using head().

In [ ]:
cd['Ratio'] = cd.Deaths/cd.Population
cd.head()

State  State Code Ten-Year Age Groups  ... Population Crude Rate     Ratio
0  Alabama           1            < 1 year  ...     3579.0        NaN  0.003912
1  Alabama           1            < 1 year  ...     7443.0      322.5  0.003225
2  Alabama           1            < 1 year  ...   169339.0     1236.0  0.012360
3  Alabama           1            < 1 year  ...   347921.0      616.2  0.006162
4  Alabama           1            < 1 year  ...     7366.0      448.0  0.004480

[5 rows x 12 columns]

3-h) Select only rows where Ten-Year Age Groups Code is equal to 1. Notice that this column has string data type.

In [ ]:
cd[cd['Ten-Year Age Groups Code']=='1']

State  State Code Ten-Year Age Groups  ... Population Crude Rate     Ratio
0     Alabama           1            < 1 year  ...     3579.0        NaN  0.003912
1     Alabama           1            < 1 year  ...     7443.0      322.5  0.003225
2     Alabama           1            < 1 year  ...   169339.0     1236.0  0.012360
3     Alabama           1            < 1 year  ...   347921.0      616.2  0.006162
4     Alabama           1            < 1 year  ...     7366.0      448.0  0.004480
...       ...         ...                 ...  ...        ...        ...       ...
4048  Wyoming          56            < 1 year  ...     3260.0        NaN  0.005828
4049  Wyoming          56            < 1 year  ...    57791.0      565.8  0.005658
4050  Wyoming          56            < 1 year  ...     3366.0     1218.1  0.012181
4051  Wyoming          56            < 1 year  ...     1572.0        NaN  0.006997
4052  Wyoming          56            < 1 year  ...    60874.0      681.7  0.006817

[354 rows x 12 columns]

3-i) Use the selected rows above (you can either assign to another variable or keep using dot(.) operator) to chain
with other functions (groupby, sum, sort_values) to view your data sorted from the highest Crude rate to
the smallest. This will show the infant mortality rate for all states.

In [ ]:
cd[cd['Ten-Year Age Groups Code']=='1'].groupby('State').sum().sort_values('Crude Rate', ascending=False)


State Code  Deaths  Population  Crude Rate     Ratio
State                                                                     
Mississippi                  224    7441    735284.0      7682.7  0.076828
Oklahoma                     320    7324    922219.0      6723.5  0.067236
North Carolina               296   17430   2160788.0      6442.4  0.064424
Wisconsin                    440    7909   1239313.0      6417.3  0.064173
Michigan                     208   16513   2184807.0      6219.8  0.062199
Kansas                       160    4885    709234.0      5850.7  0.058507
Arizona                       32   10153   1571511.0      5602.3  0.056023
Minnesota                    216    6477   1239873.0      5526.6  0.055265
South Dakota                 276    1440    201855.0      5499.9  0.054999
North Dakota                 228    1059    157084.0      5489.9  0.054900
Alaska                        16    1176    189233.0      5437.9  0.054377
Georgia                      104   19088   2388308.0      5294.5  0.052946
Illinois                     136   22039   3010235.0      5270.0  0.054343
Nebraska                     248    2765    459680.0      5216.3  0.054881
Arkansas                      35    5469    681641.0      5194.1  0.056345
Louisiana                    176   10362   1133812.0      5116.3  0.056783
Tennessee                    282   11741   1414671.0      5066.8  0.050667
Indiana                      108   11673   1522088.0      4969.7  0.049696
Alabama                        7    9828   1072918.0      4965.6  0.053567
Ohio                         273   20080   2571956.0      4902.0  0.052141
Colorado                      64    6950   1189964.0      4869.7  0.048696
Virginia                     408   12787   1804460.0      4848.7  0.051620
Missouri                     203    9888   1355762.0      4679.3  0.049668
Washington                   424    7672   1530292.0      4672.3  0.046723
Nevada                       256    3708    630443.0      4657.3  0.046573
Pennsylvania                 252   18281   2559629.0      4616.0  0.046159
South Carolina               315    8436   1034323.0      4588.1  0.048835
Florida                       96   26770   3808185.0      4413.4  0.045536
Maryland                     192    9892   1312757.0      4321.3  0.048075
Oregon                       328    4403    824379.0      4301.5  0.043015
Hawaii                        90    2039    317993.0      4280.3  0.042803
Kentucky                     126    6912    990831.0      4257.7  0.042578
Utah                         392    4657    891407.0      4238.0  0.042380
Texas                        384   42088   6769548.0      4176.1  0.041762
Delaware                      60    1682    196455.0      4154.5  0.048347
District of Columbia          44    1541    134706.0      4151.7  0.041518
California                    48   46975   9163634.0      4074.3  0.040744
West Virginia                216    2795    369035.0      3880.1  0.038801
Rhode Island                 264    1324    206609.0      3826.1  0.042306
Connecticut                   54    4067    706547.0      3822.7  0.038228
New Mexico                   280    2980    498539.0      3809.9  0.046086
Iowa                         114    3657    691738.0      3787.3  0.037873
New York                     288   24369   4332295.0      3709.1  0.037090
New Jersey                   204   10556   1928325.0      3642.1  0.036421
Montana                      180    1310    209748.0      3117.8  0.044941
Massachusetts                150    6407   1339219.0      3116.4  0.031165
Maine                        138    1410    238961.0      2571.2  0.037078
Wyoming                      280     813    126863.0      2465.6  0.037482
New Hampshire                198    1173    246082.0      2407.5  0.029367
Idaho                        128    2415    403288.0      2216.2  0.040563
Vermont                      100     542    106419.0      1013.1  0.010131